spmd/tensor/api.py

```diff
@@ -12,6 +12,62 @@
 """
 _DEBUG_STRICT = False
 
+# NOTE [Autograd interaction between torch.Tensor]
+#
+# The autograd functions defined below are being used by the public
+# facing APIs (i.e. from_local, local_tensor) to ensure our Tensor
+# works together with torch.Tensor within autograd engine. This
+# allows DistributedTensor to exist on part of the module hierarchy
+# and still able to calculate gradients across the torch.Tensor and
+# DistributedTensor boundary.
+# As an example, it enables backward computation of the following:
+#   Module A (with all torch.Tensor params)
+#       -> Module B (with all DistributedTensor params)
+#           -> Module C (with all torch.Tensor params)
+class ToTorchTensor(torch.autograd.Function):
+    @staticmethod
+    def forward(ctx, input: "Tensor"):  # type: ignore
+        ctx.previous_placement = input.placements
+        ctx.previous_device_mesh = input.device_mesh
+        return input._local_tensor
```

Comment:
The returned object here should be a "fresh" Tensor. You should add an op (.detach or .view_as) here to prevent any weird stuff.